app/src/main/java/com/kks/nimblesurveyjetpackcompose/repo/home/HomeRepoImpl.kt

```diff
@@ -0,0 +1,63 @@
+package com.kks.nimblesurveyjetpackcompose.repo.home
+
+import com.kks.nimblesurveyjetpackcompose.cache.SurveyDao
+import com.kks.nimblesurveyjetpackcompose.model.ResourceState
+import com.kks.nimblesurveyjetpackcompose.model.entities.Survey
+import com.kks.nimblesurveyjetpackcompose.model.response.SurveyResponse
+import com.kks.nimblesurveyjetpackcompose.model.response.UserResponse
+import com.kks.nimblesurveyjetpackcompose.model.response.toSurvey
+import com.kks.nimblesurveyjetpackcompose.network.ApiInterface
+import com.kks.nimblesurveyjetpackcompose.util.extensions.SUCCESS_WITH_NULL_ERROR
+import com.kks.nimblesurveyjetpackcompose.util.extensions.safeApiCall
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+import javax.inject.Inject
+
+class HomeRepoImpl @Inject constructor(
+    private val apiInterface: ApiInterface,
+    private val surveyDao: SurveyDao
+) : HomeRepo {
+
+    override fun fetchSurveyList(
+        pageNumber: Int,
+        pageSize: Int,
+        getNumberOfPage: (totalPage: Int) -> Unit
+    ): Flow<ResourceState<List<SurveyResponse>>> = flow {
+        val apiResult = safeApiCall(Dispatchers.IO) { apiInterface.getSurveyList(pageNumber, pageSize) }
+        when (apiResult) {
+            is ResourceState.Success -> {
+                apiResult.data.data?.let { surveyResponseList ->
```

Comment:
Why don't we emit anything here? 🤔

Reply:
@ryan-conway I am going to store the results and make the query from Room but I need to know is the api call is success or not to update the loading. That's why I am only emitting empty list here.

Reply:
If that's the case then why is the return type `Flow<ResourceState<List<SurveyResponse>>>`? 🤔

Reply:
Because I need to return something if the api call is a success like
`emit(ResourceState.Success(someThing()))`

Reply:
`Unit` 😉

Reply:
Ahhh that's right!! I forgot that 🤣